code/game/objects/items/RSF.dm

```diff
@@ -23,10 +23,21 @@ RSF
 	. = ..()
 	. += "<span class='notice'>It currently holds [matter]/30 fabrication-units.</span>"
 
+/obj/item/rsf/loaded
+	matter = 30
+
 /obj/item/rsf/cyborg
 	matter = 30
 
 /obj/item/rsf/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/rcd_ammo/large))
+		if((matter + 30) > 30)
+			to_chat(user, "<span class='warning'>The RSF needs to be empty in order to use a large matter cartridge on it!</span>")
+			return
```

Comment:
You should just allow use if matter is below max, let people think for themselves.